pkg/image/api/helper.go

```diff
@@ -474,6 +474,9 @@ func ImageWithMetadata(image *Image) error {
 			image.DockerImageLayers[i].Name = layer.DockerBlobSum
 		}
 		if len(manifest.History) == len(image.DockerImageLayers) {
+			// This code does not work for images imported from v2 schema
+			// registries or converted from v2 to v1, since schema v2 does not
+			// contain size information.
```

Comment:
> since schema v2 does not contain size information.

It does, it's just hidden in layer descriptors. This information is lost during conversion to v1 though.